### PR TITLE
Add calling_format support and remove gzip check.

### LINKED ISSUE #26: Smart open does not work with S3 buckets containing dots
Smart open is not working with buckets that contains `.` in name such as `com.ziky90.project`.

Particularly I am getting an error log that looks as follows.

```
Traceback (most recent call last):
  File "/home/ubuntu/bucket_accessor/model_loader.py", line 70, in <module>
    model = TModel.load(model_path)
  File "/usr/local/lib/python2.7/dist-packages/gensim/utils.py", line 252, in load
    obj = unpickle(fname)
  File "/usr/local/lib/python2.7/dist-packages/gensim/utils.py", line 906, in unpickle
    with smart_open(fname) as f:
  File "/usr/local/lib/python2.7/dist-packages/smart_open/smart_open_lib.py", line 93, in smart_open
    return S3OpenRead(parsed_uri)
  File "/usr/local/lib/python2.7/dist-packages/smart_open/smart_open_lib.py", line 191, in __init__
    self.read_key = s3_connection.get_bucket(parsed_uri.bucket_id).lookup(parsed_uri.key_id)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/connection.py", line 502, in get_bucket
    return self.head_bucket(bucket_name, headers=headers)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/connection.py", line 521, in head_bucket
    response = self.make_request('HEAD', bucket_name, headers=headers)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/connection.py", line 664, in make_request
    retry_handler=retry_handler
  File "/usr/local/lib/python2.7/dist-packages/boto/connection.py", line 1071, in make_request
    retry_handler=retry_handler)
  File "/usr/local/lib/python2.7/dist-packages/boto/connection.py", line 943, in _mexe
    request.body, request.headers)
  File "/usr/lib/python2.7/httplib.py", line 1001, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python2.7/httplib.py", line 1035, in _send_request
    self.endheaders(body)
  File "/usr/lib/python2.7/httplib.py", line 997, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 850, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 812, in send
    self.connect()
  File "/usr/lib/python2.7/httplib.py", line 1212, in connect
    server_hostname=server_hostname)
  File "/usr/lib/python2.7/ssl.py", line 350, in wrap_socket
    _context=self)
  File "/usr/lib/python2.7/ssl.py", line 566, in __init__
    self.do_handshake()
  File "/usr/lib/python2.7/ssl.py", line 796, in do_handshake
    match_hostname(self.getpeercert(), self.server_hostname)
  File "/usr/lib/python2.7/ssl.py", line 269, in match_hostname
    % (hostname, ', '.join(map(repr, dnsnames))))
CertificateError: hostname 'com.ziky90.project.s3.amazonaws.com' doesn't match either of '*.s3.amazonaws.com', 's3.amazonaws.com'
```

**NOTE:** (I use smart_open from gensim)

It seems to me that the bug is related to https://github.com/boto/boto/issues/2836 and I am wondering if for example replacing boto by boto3  would help?

### PR DESCRIPTION
**Pull request by balihoo-dengstrom**
Resolves #26 and #82

### CODE REVIEW

File: smart_open/smart_open_lib.py [hunk 1]
```diff
@@ -126,13 +127,23 @@ def smart_open(uri, mode="rb", **kw):
             if not host:
                 host = boto.config.get('s3', 'host', 's3.amazonaws.com')
 
+            calling_format = kw.pop('calling_format', None)
```
Comment (tmylk) [inline smart_open/smart_open_lib.py]: Should it be `OrdinaryCallingFormat` instead of None?

File: smart_open/smart_open_lib.py [hunk 2]
```diff
@@ -551,8 +562,8 @@ def __init__(self, outkey, min_part_size=S3_MIN_PART_SIZE, **kw):
         if not hasattr(outkey, "bucket") and not hasattr(outkey, "name"):
             raise TypeError("can only process S3 keys")
 
-        if is_gzip(outkey.name):
-            raise NotImplementedError("streaming write to S3 gzip not supported")
+        # if is_gzip(outkey.name):
```
Comment (tmylk) [inline smart_open/smart_open_lib.py]: Removing the check is a temporary solution while waiting for proper fix described by @mpenkov in #82 for all combinations of `mode="r/w", native_gzip=True/False`

**Comment by tmylk** [comment]
Thanks for the PR. Please see inline comments.

Also please add changelog and tests. We really need tests for the gzip so we don't reproduce #82 .

File: smart_open/smart_open_lib.py (hunk 2 above)
Comment (tmylk) [inline smart_open/smart_open_lib.py]: Raising the exception is correct behaviour now

**Comment by tmylk** [comment]
Ping @balihoo-dengstrom

**Comment by tmylk** [comment]
Contains #53

**Comment by menshikh-iv** [comment]
Ping @balihoo-dengstrom, what's status of this PR?

**Comment by balihoo-dengstrom** [comment]
Sorry, this can be closed. I've migrated to a different library.

**Pull request closed without merging**